Dockerfile

```diff
@@ -42,7 +42,15 @@ ENV NVIDIA_DRIVER_CAPABILITIES ${NVIDIA_DRIVER_CAPABILITIES:+$NVIDIA_DRIVER_CAPA
 # Avoid interactive prompts during the building of this docker image
 ARG DEBIAN_FRONTEND="noninteractive"
 
-RUN apt-get update && apt-get install -y lsb-release && apt-get clean ALL
+# Handle new NVIDIA GPG Keys Per https://forums.developer.nvidia.com/t/notice-cuda-linux-repository-key-rotation/212772 and https://github.com/NVIDIA/nvidia-docker/issues/1631#issuecomment-1112888486
+RUN rm /etc/apt/sources.list.d/cuda.list \
```

Comment:
Actually, thinking this through, I don't think this will work. I was thinking that these lists are package lists, but they're repository lists I believe. In that case, we should probably temporarily rename these list files so apt wont see them, do this fix, then restore the list files.

Reply:
Though, given the prevalence of this recommendation in the linked issue, maybe the cuda-keying package sets these back up on its own?

Reply:
It does, in fact setup a /etc/apt/sources.list.d/cuda-ubuntu2004-x86_64.list but no nvidia-ml.list as far as I can tell.

Reply:
If you remove the deletion of the ml- sources unfortunately, the installation still fails on apt-get update. I'll see if I can find a corresponding keyring for that

Reply:
Unfortunately, I haven't yet been able to find a new key for the ml sources, so I'm not sure what the correct next steps would be.

Reply:
It seems the ml repos are currently still using the old key. So I have added a change to stash the sources, then add the key back for only the ml repos. That seems to get us through the apt-get update steps and the image seems to be building. Hopefully it stabilizes CI as well.